src/wp-includes/fonts.php

```diff
@@ -131,7 +131,44 @@ function wp_get_font_dir() {
 	 *
 	 * @param array $defaults The original fonts directory data.
 	 */
-	return apply_filters( 'font_dir', $defaults );
+	$font_dir = apply_filters( 'font_dir', $default_dir );
```

Comment:
This needs to be applied with the final default directory known.

While the use-case we've seen is to change the default directory, there are other use cases that need to be considered. Once such example is wishing to store different font families in sub-directories, eg fonts/open-sans, fonts/source-code-pro.

It also presents the problem that WordPress may override the changes introduced by a plugin. If the directory defined by the plugin is not writable then it's the plugin's problem to handle.

Reply:
+1 I agree. 

- First, determine where the `fonts` directory should be created.
- Then pass that "final" location to the filter.
- Then use that filtered location to make the directory.

Reply:
As @peterwilsoncc noted, the first step of determining the "where" needs to include more conditionals, such as checking if the parent is writable.